scripts/jenkins/cloud/ansible/roles/list_packages/tasks/list-packages.yml

```diff
@@ -15,33 +15,74 @@
 #
 ---
 
-- name: Collect list of installed packages in plaintext
+- name: Collect list of installed packages with infos from rpm
   shell: |
-    rpm -qa | sort
+    echo "<?xml version='1.0'?>"
+    echo "<stream>"
+    rpm -qa --qf '<solvable name="%{NAME}" version="%{VERSION}" release="%{RELEASE}" disturl="%{DISTURL}"/>\n'
+    echo "</stream>"
   args:
     warn: False
   delegate_to: "{{ item_virtual_hosts }}"
   loop: "{{ [cloud_env]+groups['cloud_virt_hosts']|flatten(levels=1) }}"
   loop_control:
     loop_var: item_virtual_hosts
-  register: result_packages_plain
+  register: _result_packages_rpm
+
+- name: Collect list of installed packages repo of origin from zypper
+  shell: |
+    zypper --xmlout se -s -i -t package
+  args:
+    warn: False
+  delegate_to: "{{ item_virtual_hosts }}"
+  loop: "{{ [cloud_env]+groups['cloud_virt_hosts']|flatten(levels=1) }}"
+  loop_control:
+    loop_var: item_virtual_hosts
+  register: _result_packages_repos_zypper
+
+- name: Get Disturl and versions from rpm yml output
+  parse_xml:
+    path: "{{ item_virtual_hosts.stdout }}"
+    schema: "rpm"
+  register: _result
+  delegate_to: localhost
+  loop: "{{ _result_packages_rpm.results }}"
+  loop_control:
+    loop_var: item_virtual_hosts
+    label: "{{ item_virtual_hosts._ansible_item_label }}"
+
+- name: Get Repo of origin from zypper yml output
+  parse_xml:
+    path: "{{ item_virtual_hosts.stdout }}"
+    schema: "zypper"
+  register: _result2
+  delegate_to: localhost
+  loop: "{{ _result_packages_repos_zypper.results }}"
+  loop_control:
+    loop_var: item_virtual_hosts
+    label: "{{ item_virtual_hosts._ansible_item_label }}"
+
+- set_fact:
+    _merged_dicts_of_data: "{{ _merged_dicts_of_data | default({}) | combine(_result2.results | selectattr('_ansible_item_label', 'equalto', item._ansible_item_label) | map(attribute='parse_xml') | list | combine(item.parse_xml, recursive=True), recursive=True) }}"
+  register: _final_merged_values
+  loop: "{{ _result.results }}"
+  loop_control:
+    label: "{{ item._ansible_item_label }}"
+
+- debug: "var=item_virtual_hosts"
```

Comment:
If I understand correctly, this ansible task would print out large amounts of data, already duplicated in the files that you're generating below. Please refrain from committing this, because it would just pollute the ansible output with redundant information.